`SMTIgnoreOS` test setting breaks SMTChecker testing on OSes other than Windows, macOS and Linux

## Description

Trying to compile on OpenBSD the `develop` branch advances up to 76% but stops with:

```
[ 76%] Building CXX object test/CMakeFiles/soltest.dir/libsolidity/SMTCheckerTest.cpp.o
/home/vtamara/comp/c++/solidity/test/libsolidity/SMTCheckerTest.cpp:123:26: error: unused variable 'os' [-Werror,-Wunused-variable]
        for (std::string const& os: ignoreOSSetting | ranges::views::split(',') | ranges::to<std::vector<std::string>>())
                                ^
1 error generated.
*** Error 1 in . (test/CMakeFiles/soltest.dir/build.make:1098 'test/CMakeFiles/soltest.dir/libsolidity/SMTCheckerTest.cpp.o': cd /home/vtama...)
*** Error 2 in . (CMakeFiles/Makefile2:785 'test/CMakeFiles/soltest.dir/all': make -s -f test/CMakeFiles/soltest.dir/build.make test/CMakeFi...)
*** Error 2 in /home/vtamara/comp/c++/solidity/build (Makefile:136 'all': make -s -f CMakeFiles/Makefile2 all)
```

## Environment

- Compiler version: clang version 16.0.6
- Target EVM version (as per compiler settings): 
- Framework/IDE (e.g. Truffle or Remix): 
- EVM execution environment / backend / blockchain client:
- Operating system: OpenBSD/adJ 7.6b1

## Steps to Reproduce

```
git clone --recursive https://github.com/ethereum/solidity.git
cd solidity
mkdir build
cd build
cmake ..
make
```

--- a/test/libsolidity/SMTCheckerTest.cpp
+++ b/test/libsolidity/SMTCheckerTest.cpp
@@ -131,6 +131,10 @@ SMTCheckerTest::SMTCheckerTest(std::string const& _filename):
 #elif __linux__
 		if (os == "linux")
 			m_shouldRun = false;
+#else
+		// On other operating systems this setting is ignored (as we don't test other operating systems in CI),
+		// but we need to prevent an unused-variable warning.
+		(void)os;
 #endif
 	}
 

